### DISCUSSION TITLE
Support multiline environment variables

### DISCUSSION THREAD

**Discussion by fr4ngus** [community/community#26464] [Actions]
<p>Hi,</p>
<p>I want to store in an env variable a multiline string.</p>
<p>For instance:</p>
<pre><code>export LIST=$(find dir/* -mindepth 2 -type f -name searchname -printf '%h\n')
echo "::set-env name=LIST::$LIST"
</code></pre>
<p>So I can use for instance with xargs to loop through all line and execute a command.</p>

**Comment by BrightRan** [comment, marked as answer]
<p><a class="mention" href="/fr4ngus">@fr4ngus</a> ,</p>
<p>GitHub has supported multiline environment variables.</p>
<p>For example:</p>
<pre><code>- name: Test multiline environment variables
        run: |
          eval "$MULT_LINE_ENV01"
          echo "$MULT_LINE_ENV02"
        env:
          MULT_LINE_ENV01: |
            echo "$PWD"
            ls
            echo "command03"
          MULT_LINE_ENV02: |
            Line 01
            Line 03
            Line 02
</code></pre>
<p>Want you need to understand:</p>
<ol>
<li>When the value of a multiline environment variable are some command lines, if you want to exxcute these command lines, you can use  <strong>eval</strong> command to call this multiline environment variable.</li>
</ol>
<p>2) When the value of a multiline environment variable are some strings, you can use <strong>echo</strong>  command to call this multiline environment variable.</p>

**Comment by ChristophS-md** [comment]
@BrightRan You are showing environment variables with static values. However, @fr4ngus obviously wanted to compute them in a shell step.

**Reply by airtower-luna** [reply]
Check the date. The answer is *way* outdated, the `set-env` command has long been removed due to security issues. :wink: 

If you have a similar issue, check [how to use `GITHUB_ENV`](https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#environment-files), which natively supports multiline variables.

**Comment by AllanOricil** [comment]
I have a use case for this. 
I have a job step that outputs some json that I need to parse into a table, and finally add the resulting string to a PR as comment.  I tried to do it as shown below, but it didnt work because I can't store mult line strings to GITHUB_ENV.

![image](https://user-images.githubusercontent.com/55927613/192007725-cbb7dc53-2163-45a1-8c97-c7b3d247ab66.png)

I needed to save this string to an env so that I could pass its value to another action. There is no other way to do it, unless I create my own action to add a comment on a PR.

**Reply by airtower-luna** [reply]
There is a special syntax for writing multiline strings to `GITHUB_ENV`, see: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#multiline-strings

**Reply by AllanOricil** [reply]
I was able to implement it :D
While using the string substitution hack, I had an idea for your action. It would be cool if your action could accept a file path instead of a value stored in memory. This I could write to a file the string I want add to a PR without changing any strings.

````yml
with:
    file: ./random/dir/file.txt
    GITHUB_TOKEN: ${{ GITHUB_TOKEN }}
````
